improves metadata from pyproject.toml

maurerle opened the pull request:
## Summary of the discussion

improves metadata from pyproject.toml

- supported python versions are implicitly generated from requires-python
- license identifier makes license better visible, as it uses the common term on the https://pypi.org/project/open-mastr/ page.

## Workflow checklist

### PR-Assignee
- [ ] 🐙 Follow the workflow in [CONTRIBUTING.md](https://github.com/OpenEnergyPlatform/open-MaStR/blob/production/CONTRIBUTING.md)
- [ ] 📝 Update the [CHANGELOG.md](https://github.com/OpenEnergyPlatform/open-MaStR/blob/production/CHANGELOG.md)
- [ ] 📙 Update the documentation

### Reviewer
- [ ] 🐙 Follow the [Reviewer Guidelines](https://github.com/OpenEnergyPlatform/open-MaStR/blob/production/CONTRIBUTING.md#40-let-someone-else-review-your-pr)
- [ ] 🐙 Provided feedback and show sufficient appreciation for the work done

maurerle committed:
improves metadata from pyproject.toml

- supported python versions are implicit generated from requires-python
- license identifier makes license better visible

FlorianK13 commented:
Thanks @maurerle for this PR. We were not too familiar with the `pyproject.toml` syntax when we recently replaced our `setup.py`, so these simplifications look great. I'll test it within this week.

FlorianK13 committed:
Add license classifier #525

Based on this instruction:
https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license

FlorianK13 committed:
Update Changelog #525

FlorianK13 commented:
Tests are failing since API credentials are not provided for PRs coming from outside. I will merge anyway and rely on the tests that run before the next release.
